pkg/util/lifted/objectwatcher.go

```diff
@@ -84,5 +84,10 @@ func objectMetaObjEquivalent(a, b metav1.Object) bool {
 	if !reflect.DeepEqual(aLabels, bLabels) && (len(aLabels) != 0 || len(bLabels) != 0) {
 		return false
 	}
+	aAnnotations := a.GetAnnotations()
+	bAnnotations := b.GetAnnotations()
+	if !reflect.DeepEqual(aAnnotations, bAnnotations) && (len(aAnnotations) != 0 || len(bAnnotations) != 0) {
+		return false
+	}
```

Comment:
I think we don't need to add the judgment of Annotation, and I even think we can remove the judgment of label. Because this method is used to determine whether we need to synchronize control plane resources back to member clusters during the state collection process. It is very likely that controllers in member clusters will handle annotation and labels: 

https://github.com/karmada-io/karmada/blob/2d22b74dfce67e4f467fcfb02742cf4bfe77669b/pkg/util/objectwatcher/objectwatcher.go#L111-L117

Therefore, in most cases, this method will return true but actually it doesn't require synchronization, which wastes some computation.

Reply:
> cher/objectwatcher.go

Yea, thank you for your careful review. Add I need to delete the compare of labeles in this pr?

Reply:
No need, it can be handled by another PR.